core/src/main/java/feast/core/service/AccessManagementService.java

```diff
@@ -75,4 +85,48 @@ public void archiveProject(String name) {
   public List<Project> listProjects() {
     return projectRepository.findAllByArchivedIsFalse();
   }
+
+  /**
+   * List all members registered in the project
+   *
+   * @return List of users in the project
+   */
+  @Transactional
+  public Set<User> listMembers(String project_name) {
+    Optional<Project> project = projectRepository.findById(project_name);
+    if (!project.isPresent()) {
+      throw new IllegalArgumentException(String.format("Could not find project: \"%s\"", project_name));
+    }
+    Project p = project.get();
+    Set<User> result = p.getProjectMembers();
+    return result;
+  }
+
+  @Transactional
+  public void addMember(String user_name, String project_name) {
+    Optional<Project> project = projectRepository.findById(project_name);
+    Project p = project.get();
+
+    if (userRepository.existsUserByName(user_name)) {
+      throw new IllegalArgumentException(String.format("User already exists: %s", user_name));
```

Comment:
I wonder if this should be idempotent perhaps. Or at least not lead to an exception if the user has already been added. What do you think @gracechristina?

Reply:
Yes, agree. I think it is not needed anymore as the check of user name's uniqueness in the database has been done in the user model now.